docs/en/index.rst

```diff
@@ -13,24 +13,19 @@ and implement the following three abstract methods.
     /**
      * Lexical catchable patterns.
      *
-     * @return array
+     * @return string[]
      */
-    abstract protected function getCatchablePatterns();
+    abstract protected function getCatchablePatterns(): array;
```

Comment:
This code block mirrors the code from `AbstractLexer`. We should not add native types here until we've added them to `AbstractLexer` as well.